File: problems/3427-sum-of-variable-length-subarrays/solution.java
class Solution {
    public int subarraySum(int[] nums) {
        int n = nums.length;
        long totalSum = 0;
        for (int i = 0; i < n; i++) {
            int start = Math.max(0, i - nums[i]);
            long subarraySum = 0;
            for (int j = start; j <= i; j++) {
                subarraySum += nums[j];
            }
            totalSum += subarraySum;
        }
        return (int) totalSum;
    }
}

File: problems/3427-sum-of-variable-length-subarrays/solution.py
class Solution:
    def subarraySum(self, nums: List[int]) -> int:
        total_sum = 0
        for i in range(len(nums)):
            start = max(0, i - nums[i])
            sub_array_sum = sum(nums[start:i+1])
            total_sum += sub_array_sum
        return total_sum